lib/SILOptimizer/Transforms/PhiArgumentOptimizations.cpp

```diff
@@ -111,8 +157,49 @@ bool RedundantPhiEliminationPass::optimizeArgs(SILBasicBlock *block) {
         continue;
 
       if (valuesAreEqual(arg1, arg2)) {
-        arg2->replaceAllUsesWith(arg1);
-        erasePhiArgument(block, arg2Idx);
+        if (block->getParent()->hasOwnership()) {
+          auto *phi1 = cast<SILPhiArgument>(arg1);
+          auto *phi2 = cast<SILPhiArgument>(arg2);
+          // @owned phi args can only be equal if all the incoming values had a
+          // None ownership. To replace, create a copy_value of the duplicate
+          // arg.
+          if (phi1->getOwnershipKind() == OwnershipKind::Owned &&
+              phi2->getOwnershipKind() == OwnershipKind::Owned) {
+#ifndef NDEBUG
```

Comment:
Well, for the assert you don't need the `#ifndef NDEBUG`. In NDEBUG the assert is a no-op anyway

Reply:
yeah, I'll piggyback this with another open PR.